packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -71,15 +81,20 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
   const userEnteredQuery = form.watch("query");
   const userEnteredDimensions = form.watch("dimensions");
   const userEnteredHasNameCol = form.watch("hasNameCol");
+  const userEnteredDimsForTraffic = form.watch("dimensionsForTraffic");
 
   const handleSubmit = form.handleSubmit(async (value) => {
-    await onSave(value);
+    // Do a JIT migration if user doesn't supply any dimensions for traffic
```

Comment:
Sorry for he back and forth but IDK if we should do this if we're going to choose to update the defaults later (e.g. in a couple of months). So maybe instead just leave it undefined.

Reply:
Will do